### PR TITLE
Refactor polynomials

### PR DESCRIPTION
**Pull request by hemalvarambhia**
- Improved naming;
- Tightened Encapsulation;
- Added some missing tests, including demonstrating that multiplication is commutative.
- Made tests more like examples written with DSLs

### CODE REVIEW

**Comment by hernanmd** [comment]
There is only 1 test failing `testOptimize` on linux

**Comment by hemalvarambhia** [comment]
> There is only 1 test failing `testOptimize` on linux
> 

Restarted the CI run.

File: src/Math-Polynomials/PMPolynomial.class.st [hunk 1]
```diff
@@ -239,20 +239,17 @@ PMPolynomial >> roots [
 { #category : #information }
 PMPolynomial >> roots: aNumber [
 
-	| pol roots root rootFinder |
+	| polynomial roots root rootFinder |
 	rootFinder := PMNewtonZeroFinder with: aNumber.
```
Comment (hemalvarambhia) [inline src/Math-Polynomials/PMPolynomial.class.st:243]: Is `aNumber` the starting point for computing the root i.e. 
$`x_0`$

Comment (hernanmd) [reply src/Math-Polynomials/PMPolynomial.class.st:243]: Is that a question?

Comment (hemalvarambhia) [reply src/Math-Polynomials/PMPolynomial.class.st:243]: Yes. It is a question.

Comment (hernanmd) [reply src/Math-Polynomials/PMPolynomial.class.st:243]: aNumber is the initial guess for the root (in PMNewtonZeroFinder x0 is the initial guess for the root, which is passed to the `initValue:` method, which sets the initial value for the iterations)

**Review by hernanmd — APPROVED** [review]

**Pull request merged**